### ISSUE TITLE
Requiring dependencies before installing shinyapp

### ISSUE THREAD

**Issue by ck2136** [AdamSpannbauer/r_regex_tester_app#10]
Hi!

Thank you fro a great app. I just cloned the repo and had to go through installing dependencies. I think it might be worth packaging the app up via using the `golem` pacakge. I'll try to see if I can get a pull request going soon.

**Comment by AdamSpannbauer** [comment]
Hi @ck2136, 

I would gladly review a PR for this, and I think doing the refactor would have a lot of benefits to the project's health.  Please ask any questions as they come up and I'll do my best to help.  Unfortunately, the code is a mess that I'm not too proud of, so it might take some time for me to provide answers.  Good luck!

**Comment by ck2136** [comment]
@AdamSpannbauer 
Thanks for the response! I don't think I'm too experienced of a coder either but I think it'd be worth a shot to package this up as it seems to be short and sweet enough. I've made a bunch of files and made a pull request. Essentially these changes are all based on pulling your original code into the `golem` shiny app packaging framework.  I recently made a shiny app using it and think it's valuable to add unit tests to make sure the app produces the desired outputs. Let me know what you think!
